DEVDOCS-1975: Building Apps Introduction (#514)
bigcommerce/dev-docs: docs/api-docs/apps/guide/apps-01-introduction.md

# Introduction to Building Apps

<div class="otp" id="no-index">

> ### On This Page

Reviewer: The section headings here and throughout the article should be in [sentence case](https://intranet.bigcommerce.com/display/SST/Headings+and+Titles)

- [About BigCommerce Apps](#about-bigcommerce-apps)
- [Use Cases](#use-cases)
- [Getting Started](#getting-started)
- [Next Steps](#next-steps)
- [Resources](#resources)

</div>

> Using BigCommerce's powerful APIs, SDKs, and Toolkits, developers can build apps that are installable on BigCommerce stores. Once approved, apps can be sold (or made available free of charge) to all BigCommerce merchants on the [Apps Marketplace](https://www.bigcommerce.com/apps). This is the first article in a comprehensive developer's guide to building BigCommerce apps. It provides a brief, high-level introduction for developers new to the platform. If you already have experience developing for BigCommerce, feel free to [skip ahead](#next-steps) or checkout the [quick start tutorial](https://developer.bigcommerce.com/api-docs/apps/quick-start).

Reviewer: ```suggestion
Using BigCommerce's powerful APIs, SDKs, and toolkits, developers can build apps that are installable on BigCommerce stores. Once approved, apps can be sold (or made available free of charge) to all BigCommerce merchants on the [Apps Marketplace](https://www.bigcommerce.com/apps). This is the first article in a comprehensive developer's guide to building BigCommerce apps. It provides a brief, high-level introduction for developers new to the platform. If you already have experience developing for BigCommerce, feel free to [skip ahead](#next-steps) or check out the [quick start tutorial](https://developer.bigcommerce.com/api-docs/apps/quick-start).

```

## About BigCommerce Apps

Let's first take a look how apps are discovered, displayed, and managed on BigCommerce stores.

### Discovery

> Approved apps are listed on the app [Marketplace](https://www.bigcommerce.com/apps/) for merchants to browse, search, and install:

Reviewer: ```suggestion
Approved apps are listed on the app [Marketplace](https://www.bigcommerce.com/apps/) for merchants to browse, search, and install.

```

![App Marketplace](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-01-introduction-01.png "App Marketplace")

### Management

> Apps installed on a store are managed in the store's control panel under [**Apps** > My **Apps**](https://login.bigcommerce.com/deep-links/manage/marketplace/apps/my-apps/drafts). Here, store users can launch or uninstall the app:

Reviewer: ```suggestion
Apps installed on a store are managed in the store's control panel under **Apps** > [**My Apps**](https://login.bigcommerce.com/deep-links/manage/marketplace/apps/my-apps/drafts). Here, store users can launch or uninstall the app.

```

![App Management](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-01-introduction-02.png  "App Management")

### Display

> When an launches an app, BigCommerce makes a request to the app's `/load` callback and displays the app's UI in an iFrame:

Reviewer: ```suggestion
When an launches an app, BigCommerce makes a request to the app's `/load` callback and displays the app's UI in an iFrame.

```

> When an launches an app, BigCommerce makes a request to the app's `/load` callback and displays the app's UI in an iFrame:

Reviewer: Unclear what "When an launches an app," means in this context.

![App Display](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-01-introduction-03.png  "App Display")

## Use Cases
What can developers do with BigCommerce apps?

### Integrate with Business Systems

> Use [BigCommerce APIs](https://developer.bigcommerce.com/api-reference) to integrate stores with business systems such as:

Reviewer: ```suggestion
Use [BigCommerce APIs](https://developer.bigcommerce.com/api-reference) to integrate stores with business systems such as the following:

```

Reviewer: [Colons](https://intranet.bigcommerce.com/display/SST/Colons)

* [ERP systems](https://www.bigcommerce.com/apps/erp/)
* [Oder management platforms](https://www.bigcommerce.com/apps/catalog-order-management/)
* [Accounting software](https://www.bigcommerce.com/apps/accounting-tax/)
* [Shipping and Fulfillment services](https://www.bigcommerce.com/apps/shipping-fulfillment/)

> [Learn more about BigCommerce's APIs](https://developer.bigcommerce.com/api-reference)

Reviewer: ```suggestion
[Learn more about BigCommerce's APIs](https://developer.bigcommerce.com/api-reference).

```

### Connect to Sales Channels
List products on external [sales channels](https://www.bigcommerce.com/apps/sales-channels/) with [Channels Toolkit](https://developer.bigcommerce.com/api-docs/channels/channels-toolkit-reference).

[Learn more about connecting apps to sales channels.](https://developer.bigcommerce.com/api-docs/channels/overview).

### Add and Manage Scripts
Create and manage scripts with [Scripts API](https://developer.bigcommerce.com/api-reference/store-management/scripts) and [Script Manager](https://support.bigcommerce.com/s/article/Using-Script-Manager).

[Learn more about creating scripts](https://developer.bigcommerce.com/api-docs/storefront/scripts-overview).

### Create Custom Widgets
Create [widgets](https://support.bigcommerce.com/s/article/Page-Builder#builder) for merchants to install and configure.

[Learn more about creating widgets](https://developer.bigcommerce.com/api-docs/storefront/widgets/widgets-overview).

## Getting Started
Here's how to get started with BigCommerce development:

> 1. [Start a Free Trial](https://www.bigcommerce.com/essentials/free-trial/) to test apps against.

Reviewer: ```suggestion
1. [Start a free trial](https://www.bigcommerce.com/essentials/free-trial/) to test apps against.

```

> 2. [Apply to be a Partner](https://www.bigcommerce.com/partners) to get payed for your apps.

Reviewer: ```suggestion
2. [Apply to be a Partner](https://www.bigcommerce.com/partners) to get paid for your apps.

```

> 3. [Create a DevTools Account](https://devtools.bigcommerce.com/) to register and submit apps.

Reviewer: ```suggestion
3. [Create a DevTools account](https://devtools.bigcommerce.com/) to register and submit apps.

```

4. [Join the Developer Community](https://support.bigcommerce.com/s/group/0F913000000HLjECAW/bigcommerce-developers) to get help from other BigCommerce developers.

> 5. [Checkout the Marketplace](https://www.bigcommerce.com/marketplace/) to get inspiration on what to build.

Reviewer: ```suggestion
5. [Check out the Marketplace](https://www.bigcommerce.com/marketplace/) to get inspiration on what to build.

```

## Next Steps

> * [Decide Type of App to Build](https://developer.bigcommerce.com/api-docs/apps/guide/types-of-apps)

Reviewer: ```suggestion
* [Decide which type of app to build](https://developer.bigcommerce.com/api-docs/apps/guide/types-of-apps)

```

> * [Get Started Building Apps with CodeSandbox, Node, and Express](https://developer.bigcommerce.com/api-docs/apps/quick-start)

Reviewer: ```suggestion
* [Start building apps with CodeSandbox, Node, and Express](https://developer.bigcommerce.com/api-docs/apps/quick-start)

```
